packages/date-parser/src/helpers/utcParsingHelpers.js

```diff
@@ -0,0 +1,45 @@
+import convertTimezone from './convertTimezone';
+import luxonFromStruct from './luxonFromStruct';
+import { DATE_UNIT_LEVELS } from '../constants';
+import dateStructFromLuxon from './dateStructFromLuxon';
+
+const shouldUseUTC = (dateUnit) => {
+  return DATE_UNIT_LEVELS[dateUnit] >= DATE_UNIT_LEVELS.hour;
+};
+
+/**
+ *
+ * @param {Date} ref
+ * @param {DATE_UNIT_LEVELS} dateUnit
+ * @param {string} timezone
+ * @returns {Date}
+ */
+const adjustRefdate = (ref, dateUnit, timezone) => {
+  if (!timezone) { return ref; }
+
+  // If the computed unit > hour, then we don't care about DST, we can adjust the clock from UTC to wallclock BEFORE CALCULATION
+  //
+  // Otherwise, we should stay with UTC because calculating in wallclock when DST happens is way to complex.
+  // AFTER CALCULATION we offset the UTC -> wallclock again
+  //
+  // In short:
+  //           unit >  hour: Adjust the ref   -> calculate
+  //           unit <= hour: calculate        -> Adjust the result
+  //
+  // Related reading: https://moment.github.io/luxon/index.html#/math?id=calendar-math-vs-time-math
```

Comment:
this is really cool. I'm seeing whether we can refactor to follow this logic and make it clear whether we are doing calendar math or time math